## ПРАКТИЧНА РОБОТА 3.2

### Варіант 15

Завдання 1: Скачайте датасет  Amazon Top 50 Bestselling Books 2009 – 2019, що знаходиться  за посиланням.Прочитайте csv файл (метод read_csv).

In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

In [2]:
books = pd.read_csv("bestsellers with categories.csv")

print(books)

                                                  Name  \
0                        10-Day Green Smoothie Cleanse   
1                                    11/22/63: A Novel   
2              12 Rules for Life: An Antidote to Chaos   
3                               1984 (Signet Classics)   
4    5,000 Awesome Facts (About Everything!) (Natio...   
..                                                 ...   
545       Wrecking Ball (Diary of a Wimpy Kid Book 14)   
546  You Are a Badass: How to Stop Doubting Your Gr...   
547  You Are a Badass: How to Stop Doubting Your Gr...   
548  You Are a Badass: How to Stop Doubting Your Gr...   
549  You Are a Badass: How to Stop Doubting Your Gr...   

                       Author  User Rating  Reviews  Price  Year        Genre  
0                    JJ Smith          4.7    17350      8  2016  Non Fiction  
1                Stephen King          4.6     2052     22  2011      Fiction  
2          Jordan B. Peterson          4.7    18979     15  201

Завдання 2: Виведіть перші 10 рядків (метод head)

In [3]:
books10 = books.head(10) 

print(books10) 

                                                Name  \
0                      10-Day Green Smoothie Cleanse   
1                                  11/22/63: A Novel   
2            12 Rules for Life: An Antidote to Chaos   
3                             1984 (Signet Classics)   
4  5,000 Awesome Facts (About Everything!) (Natio...   
5      A Dance with Dragons (A Song of Ice and Fire)   
6  A Game of Thrones / A Clash of Kings / A Storm...   
7                     A Gentleman in Moscow: A Novel   
8      A Higher Loyalty: Truth, Lies, and Leadership   
9                          A Man Called Ove: A Novel   

                     Author  User Rating  Reviews  Price  Year        Genre  
0                  JJ Smith          4.7    17350      8  2016  Non Fiction  
1              Stephen King          4.6     2052     22  2011      Fiction  
2        Jordan B. Peterson          4.7    18979     15  2018  Non Fiction  
3             George Orwell          4.7    21424      6  2017      Fic

Завдання 3: Виведіть розміри датасету (метод shape) 

In [4]:
num_rows, num_columns = books.shape

print(f"Кількість рядків: {num_rows}, Кількість стобчиків: {num_columns}")

Кількість рядків: 550, Кількість стобчиків: 7


Питання: Про скільки книг зберігає дані датасет? Відповідь: 550 книг

Завдання 4: Змініть назви всіх колонок: назва з маленької букви, без пробілів через підкреслення. 

In [5]:
books = books.rename(columns = {
    "Name": "name",
    "Author": "author",
    "User Rating": "user_rating",
    "Reviews": "reviews",
    "Price": "price",
    "Year": "year",
    "Genre": "genre"})

print("Перейменовані назви колонок")
print(books)

Перейменовані назви колонок
                                                  name  \
0                        10-Day Green Smoothie Cleanse   
1                                    11/22/63: A Novel   
2              12 Rules for Life: An Antidote to Chaos   
3                               1984 (Signet Classics)   
4    5,000 Awesome Facts (About Everything!) (Natio...   
..                                                 ...   
545       Wrecking Ball (Diary of a Wimpy Kid Book 14)   
546  You Are a Badass: How to Stop Doubting Your Gr...   
547  You Are a Badass: How to Stop Doubting Your Gr...   
548  You Are a Badass: How to Stop Doubting Your Gr...   
549  You Are a Badass: How to Stop Doubting Your Gr...   

                       author  user_rating  reviews  price  year        genre  
0                    JJ Smith          4.7    17350      8  2016  Non Fiction  
1                Stephen King          4.6     2052     22  2011      Fiction  
2          Jordan B. Peterson      

Завдання 5: Вивести кількість пропусків у кожному стовпчику. 

In [6]:
missing_sum = books.isnull().sum()

print("Сума пропусків в кожній колонці: ")
print(missing_sum)

Сума пропусків в кожній колонці: 
name           0
author         0
user_rating    0
reviews        0
price          0
year           0
genre          0
dtype: int64



Питання : Чи є в якихось змінних пропуски?Відповідь: Ні 

Завдання 6: За допомогою методу unique  перевірте. Які є унікальні жанри в стовпці genre  

In [7]:
unique = books['genre'].unique()
print("Унікальні жанри в стовпці genre: ", unique)

Унікальні жанри в стовпці genre:  ['Non Fiction' 'Fiction']


Питання : Які є унікальні жанри? Відповідь: 'Non Fiction' 'Fiction'

Завдання 7: Визначте максимальну, мінімальну, медіанну ціну за допомогою методів  max, min, mean, median 

In [8]:
max_price = books['price'].max()
min_price = books['price'].min()
mean_price = books['price'].mean()
median_price = books['price'].median()

print("Mаксимальнa ціни: ", max_price)
print("Mінімальнa ціни: ", min_price)
print("Cередня ціна: ", mean_price)
print("Медіана ціни: ", mean_price)

Mаксимальнa ціни:  105
Mінімальнa ціни:  0
Cередня ціна:  13.1
Медіана ціни:  13.1


Питання:Максимальна ціна? Відповідь: 105
Питання: Мінімальна ціна? Відповідь: 0
Питання: Середня ціна? Відповідь: 13.1
Питання: Медіана ціни? Відповідь: 13.1

Завдання 8: Виконайте наступні завдання з пошуку і сортування даних 

In [9]:
#Питання: Який рейтинг у датасеті найвищий?  
sorted_books = books.sort_values(by = ['user_rating'], ascending = False)
max_rating = sorted_books['user_rating'].iloc[0]

print(max_rating)

4.9


In [10]:
#Питання: Скільки книг мають такий рейтинг?  
count_max_rating = (sorted_books['user_rating'] == max_rating).sum()

print("Кількість книг що мають найбільший рейтинг: ", count_max_rating)

Кількість книг що мають найбільший рейтинг:  52


In [11]:
#Питання: Яка книга має найбільше відгуків? 
max_reviews = sorted_books['reviews'].idxmax()
name_max = books.loc[max_reviews,'name']

print("Книга з найбільшою кількістю відгуків:", name_max)

Книга з найбільшою кількістю відгуків: Where the Crawdads Sing


In [12]:
#Питання: З тих книг, що потрапили до Топ-50 у 2010 році, яка книга найдорожча ?
max_price_2010 = books[(books['year'] == 2010)]['price'].idxmax()
name_max = books.loc[max_price_2010,'name']

print("Найдорожча книга за 2010 рік: ", name_max)

Найдорожча книга за 2010 рік:  Publication Manual of the American Psychological Association, 6th Edition


In [13]:
#Скільки книг жанру Fiction потрапили до Топ-50 у 2012 році 
books_fiction_2012 = (books['year'] == 2012) & ( books['genre'] == 'Fiction')
count_books = books_fiction_2012.sum()

print("Кількість книг жанру Fiction потрапили до Топ-50 у 2012 році: ", count_books)

Кількість книг жанру Fiction потрапили до Топ-50 у 2012 році:  21


In [14]:
#Питання : Скільки книг з рейтингом 4.9 потрапило до рейтингу у 2010 та 2011 роках (використовуйте | або метод isin)?  
book_rate_10_11 = (books['user_rating'] == 4.9) & ((books['year'] == 2010) | (books['year'] == 2011))
count_books_rate = book_rate_10_11.sum()
print("Кількість книг з рейтингом 4.9 потрапило до рейтингу у 2010 та 2011 роках: ", count_books_rate)

Кількість книг з рейтингом 4.9 потрапило до рейтингу у 2010 та 2011 роках:  1


Вивести максимальну і  мінімальну ціну для жанру Fiction I NonFiction (за допомогою методів  groupby та agg, для підрахунку мінімальних та максимальних значень використовуйте max та min). Використовуйте тількі потрібні вам стовпці. 

In [15]:
genre_prices = books.groupby('genre')['price'].agg(['max', 'min'])
print("Максимальна і мінімальна ціни для жанру Fiction та NonFiction(за допомогою методів  groupby та agg):")
print(genre_prices)

Максимальна і мінімальна ціни для жанру Fiction та NonFiction(за допомогою методів  groupby та agg):
             max  min
genre                
Fiction       82    0
Non Fiction  105    0


На практичній роботі я вдосконалила знання із мови програмування - Python, а саме попрактикувалася зчитувати дані з CSV файлу та працювати із ними (знаходити розмір датасету, знаходити максимальні та мінімальні значення по стовбцях, заповнювати пропуски та групувати дані по певним критеріям). 